azure/database/mssql/server.tf

```diff
@@ -0,0 +1,18 @@
+resource "azurerm_mssql_server" "managed" {
+  count = var.server.preexisting_name == null ? 1 : 0
```

Comment:
QoL: Just a thought/idea. Probably not to important.

We used `validation` in the gateway module [here](https://github.com/THEY-Consulting/they-terraform/blob/main/aws/lambda/gateway/variables.tf#L72) to result in an error when providing invalid (invalid here rather means contradicting) configuration variables. Perhaps this could also be of use somewhere in the database/mssql module. We got the docu in place, but this would verify that the user did not misconfigure the inputs.

Reply:
Added a check for `administrator_login_password` to prevent leaving both `preexisting_name` and the password blank. 

It would be nice to also throw an error if other server configs are set while `preexisting_name` is also set. This would make it obvious, that the other config values will be ignored in this case. But this would mean we couldn't set defaults anymore. At the moment I would prefer having the defaults set directly within the variable definition over improved error feedback. 

Maybe we can revisit this later if it becomes a common problem when using this module.